modules/integration/src/test/scala/scala/cli/integration/FixBuiltInRulesTestDefinitions.scala

```diff
@@ -447,4 +447,81 @@ trait FixBuiltInRulesTestDefinitions { _: FixTestDefinitions =>
+                          |  println(os.pwd)
+                          |}
+                          |""".stripMargin,
+        "script.sc" -> s"""$directive
+                          |println(os.pwd)
+                          |""".stripMargin
+      )
+      if !Properties.isWin // TODO: make this run on Windows CI
+      testInputs = TestInputs(os.rel / inputFileName -> code)
+    }
+      test(
+        s"dont extract directives into project.scala for a single-file project: $inputFileName"
+      ) {
+        testInputs.fromRoot { root =>
+          val fixResult = os.proc(TestUtil.cli, "--power", "fix", ".", extraOptions)
+            .call(cwd = root, stderr = os.Pipe)
+          expect(fixResult.err.trim().contains(
+            "No need to migrate directives for a single source file project"
+          ))
+          expect(!os.exists(root / projectFileName))
+          expect(os.read(root / inputFileName) == code)
+          val runResult = os.proc(TestUtil.cli, "run", ".", extraOptions)
+            .call(cwd = root, stderr = os.Pipe)
+          expect(runResult.out.trim() == root.toString)
+        }
+      }
+  }
+
+  if (!Properties.isWin)
+    test("all test directives get extracted into project.scala") {
+      val osLibDep               = "com.lihaoyi::os-lib:0.11.3"
+      val munitDep               = "org.scalameta::munit:1.0.4"
+      val pprintDep              = "com.lihaoyi::pprint:0.9.0"
+      val osLibDepDirective      = s"//> using dependency $osLibDep"
+      val osLibTestDepDirective  = s"//> using test.dependency $osLibDep"
+      val munitTestDepDirective  = s"//> using test.dependency $munitDep"
+      val pprintTestDepDirective = s"//> using test.dependency $pprintDep"
+      val mainFilePath           = os.rel / "Main.scala"
+      val testFilePath           = os.rel / "MyTests.test.scala"
+      TestInputs(
+        mainFilePath -> s"""$munitTestDepDirective
+                           |object Main extends App {
+                           |  def hello: String = "Hello, world!"
+                           |  println(hello)
+                           |}
+                           |""".stripMargin,
+        testFilePath -> s"""$osLibDepDirective
+                           |$pprintTestDepDirective
+                           |import munit.FunSuite
+                           |
+                           |class MyTests extends FunSuite {
+                           |  test("hello") {
+                           |    pprint.pprintln(os.pwd)
+                           |    assert(Main.hello == "Hello, world!")
+                           |  }
+                           |}
+                           |""".stripMargin
+      ).fromRoot { root =>
+        os.proc(TestUtil.cli, "--power", "fix", ".", extraOptions).call(cwd = root)
+        val projectFileContents = os.read(root / projectFileName)
+        expect(projectFileContents.contains(munitTestDepDirective))
```

Comment:
Could we compare the whole expected file?

Reply:
done